PowerShell.Common.props

```diff
@@ -1,10 +1,110 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!--
+        We should ~explicitly~ assign versions properties in the target
+        because:
+          1. By default the version properties defined in global PropertyGroup (not target!) in 'Microsoft.NET.DefaultAssemblyInfo.targets'
+          2. Properties in global 'PropertyGroups' is assigned before any targets
+          3. and cannot be conditionally excecuted as targets (after this target).
+
+        To assign versions properties we should excecute the target:
+          '_GenerateRestoreProjectSpec' - before 'Restore' target
+          'GenerateNuspec'              - before 'Pack' target
+          'BeforeBuild'                 - before 'Build' target
+  -->
+  <Target Name="GetPSCoreVersionFromGit"
+    BeforeTargets="_GenerateRestoreProjectSpec;GenerateNuspec;BeforeBuild"
+  >
+
+    <PropertyGroup>
+      <GitExe>git</GitExe>
+    </PropertyGroup>
+
+    <Exec Command='$(GitExe) rev-parse --show-cdup' ConsoleToMSBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="GitCDUP" />
+    </Exec>
```

Comment:
I think we don't need this step. When running `git describe` below, just set the `WorkingDirectory=$(MSBuildProjectDirectory)`, then it's guaranteed that the `git` command executes from the directory where `.csproj` locals, so `--git-dir` would be unnecessary.

Reply:
Yes, VS can. Mentioned CLI Issue imply about this.

Will fix.